@here/harp-mapview/lib/poi/PoiManager.ts

```diff
@@ -450,10 +394,78 @@ export class PoiManager {
             tile.tileKey.level
         );
 
-        if (textElement !== undefined) {
-            // If the poi icon is rendered, the label that shows its text should also be rendered.
-            // The distance rule of the icon should apply, not the one for text (only) labels.
-            textElement.ignoreDistance = true;
+        // If the poi icon is rendered, the label that shows text should also be rendered.
+        // The distance rule of the icon should apply, not the one for text (only) labels.
+        textElement.ignoreDistance = false;
+        tile.addTextElement(textElement);
+    }
+
+    /**
+     * Create and add POI [[TextElement]]s to tile with a series of positions.
+     */
+    private addPoi(
+        tile: Tile,
+        poiGeometry: PoiGeometry,
+        technique: PoiTechnique,
+        techniqueIdx: number,
+        positions: THREE.BufferAttribute
+    ) {
+        if (poiGeometry.stringCatalog === undefined) {
+            return;
+        }
+
+        const techniqueTextureName: string | undefined =
+            technique.imageTexture !== undefined
+                ? composeTechniqueTextureName(technique.imageTexture, technique)
+                : undefined;
+
+        const poiTechnique = technique as PoiTechnique;
+        const poiTableName = poiTechnique.poiTable;
+        let poiName = poiTechnique.poiName;
+
+        for (let i = 0; i < positions.count; ++i) {
+            const x = positions.getX(i);
+            const y = positions.getY(i);
+            const z = positions.getZ(i);
+
+            assert(poiGeometry.texts.length > i);
+            let imageTextureName;
```

Comment:
You even can do
```suggestion
            let imageTextureName = techniqueTextureName;
```

Reply:
Yeah, I was even struggling, but old way won this time :+1: